tests/helpers/yadda-annotations.js

```diff
@@ -56,4 +45,35 @@ function runScenario(featureAnnotations, scenarioAnnotations) {
   return checkAnnotations(scenarioAnnotations);
 }
 
-export { runFeature, runScenario };
+// setup tests
+// you can override these function to add additional setup setups, or handle new setup related annotations
+function setupFeature(featureAnnotations) {
+  return setupYaddaTest(featureAnnotations);
+}
+
+function setupScenario(featureAnnotations, scenarioAnnotations) {
+  let setupFn = setupYaddaTest(scenarioAnnotations);
+  if (setupFn && (featureAnnotations.application || featureAnnotations.rendering || featureAnnotations.context)) {
```

Comment:
Can we pick the most common one here? Instead of failing with an exception? Or would that cause to much headache for a new user. Maybe throw a warning, ember people have been trained to look out for those :smile:.

Reply:
Not sure I understand. This is guarding against having an Ember setup annotation in the feature as well as the scenario, as that would cause the `setup*Test()` function to be called twice (in the module/describe block of the feature *and* that of the scenario). This will cause some other (rather hard to understand) exceptions, something along the lines of "you cannot use the application div twice" , as there will be two `beforeEach` hooks called that do the same setup work (e.g. initialize application) twice.

Reply:
Oh right I misunderstood/misread, sorry 👍